### ISSUE TITLE
systemctl reload snap.lxd.daemon.service starts stopped instances with boot.auto = true

### ISSUE THREAD

**Issue by vosdev** [canonical/lxd#14464]
# Required information

 * Distribution:  Ubuntu
 * Distribution version: 24.04
 * The output of "snap list --all lxd core20 core22 core24 snapd":
 ```
 Name    Version      Rev    Tracking       Publisher   Notes
core22  20240904     1621   latest/stable  canonical✓  base,disabled
core22  20241001     1663   latest/stable  canonical✓  base
lxd     6.1-78a3d8f  30130  latest/stable  canonical✓  -
snapd   2.63         21759  latest/stable  canonical✓  snapd
```

# Issue description

Updating LXD's certificates I noticed that instances that were in a `Stopped` state, changed to `Running` after issueing `systemctl reload snap.lxd.daemon.service`.

# Steps to reproduce

 1. Stop an instance that has `boot.autostart: "true"` configured.
 2. systemctl reload snap.lxd.daemon.service
 3. The instance is now running

I did not expect it to change to Running after a reload, I only expect this after a restart.

**Comment by MggMuggins** [comment]
This also results in instances auto starting when the daemon is auto-refreshed by snapd. While [the docs](https://documentation.ubuntu.com/lxd/en/stable-5.21/reference/instance_options/#instance-boot:boot.autostart) are fairly clear that `boot.autostart` is tied to the LXD server's lifetime, I agree that this behavior is sub-optimal in a snap-deployed context.

**Comment by MggMuggins** [comment]
Yeah, it's based on the signal (`SIGTERM` for reload and `SIGPWR` for shutdown). The problem is distinguishing during start how the daemon was shut down; that is, to effectively disable `boot.autostart` when the daemon was shut down with `SIGTERM`.

**Comment by tomponline** [comment]
Wonder if we can set an env var on stop so its detectabke on start up.
